### PR TITLE
bpf: Extend datapath options for K8sVerifier test

### PR DESCRIPTION
**Pull request by pchaigno**
Please see commits.

I've checked locally that K8sVerifier passes on all three kernels (4.9, 4.19, and bpf-next). I'll rerun the full end-to-end tests here for good measure once the reviews are in.

### CODE REVIEW

**Review by qmonnet — APPROVED** [review]
Thanks for that. Looks good, and no objection from my side for the last commit.

**Comment by pchaigno** [comment]
test-only --kernel_version=4.9 --focus="K8sVerifier"

**Comment by pchaigno** [comment]
test-only --kernel_version=49 --focus="K8sVerifier"

**Comment by pchaigno** [comment]
test-only --kernel_version=netnext --focus="K8sVerifier"

**Comment by pchaigno** [comment]
test-only --kernel_version=419 --focus="K8sVerifier"

**Comment by pchaigno** [comment]
test-only --kernel_version=419 --focus="K8sVerifier"

File: bpf/Makefile [hunk 1]
```diff
@@ -92,10 +92,13 @@ MAX_BASE_OPTIONS += -DHAVE_LPM_TRIE_MAP_TYPE=1 -DHAVE_LRU_HASH_MAP_TYPE=1 \
 	-DENABLE_EXTERNAL_IP=1 -DENABLE_NODEPORT_ACCELERATION=1 \
 	-DENABLE_SESSION_AFFINITY=1 -DENABLE_DSR_ICMP_ERRORS=1 -DENABLE_DSR=1 \
 	-DENABLE_HOSTPORT=1 -DENABLE_LOADBALANCER -DENABLE_DSR_HYBRID=1 \
-	-DENABLE_DSR_ICMP_ERRORS=1 -DENABLE_IPV4_FRAGMENTS=1 -DENABLE_CUSTOM_CALLS=1
+	-DENABLE_DSR_ICMP_ERRORS=1 -DENABLE_IPV4_FRAGMENTS=1
 ifeq ("$(KERNEL)","netnext")
+# We need to define ENABLE_HOST_SERVICES_FULL to work around complexity issue #14234.
+# We define ENABLE_CUSTOM_CALLS only for net-next (vs. net-next and 4.19) to work
```
Comment (joestringer) [inline bpf/Makefile:98]: only for net-next vs net-next? ;)

**Review by joestringer — APPROVED** [review]

File: bpf/Makefile (hunk 1 above)
Comment (pchaigno) [reply bpf/Makefile:98]: Only net-next vs. net-next + 4.19.

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-gke

**Comment by pchaigno** [comment]
k8s-1.20-kernel-4.19 failed with known flake https://github.com/cilium/cilium/issues/15337. Other CI jobs are green and reviews are in. Merging.

**Pull request merged**